File: solutions/JavaScript/374.GuessNumberHigherOrLower.js
var guessNumber = function (n) {
    let left = 1;
    let right = n;
    while (left <= right) {
        let mid = Math.floor((left + right) / 2);
        let pick = guess(mid);
        if (pick === -1) {
            right = mid - 1;
        } else if (pick === 1) {
            left = mid + 1;
        } else {
            return mid;
        }
    }
};

File: solutions/Python/374.GuessNumberHigherOrLower.py
class Solution:
    def guessNumber(self, n: int) -> int:
        left: int = 1
        end: int = n

        while left <= end:
            mid: int = (left + end) // 2
            pick = guess(mid)
            if pick == -1:
                end = mid - 1
            elif pick == 1:
                left = mid + 1
            else:
                return mid
